### ISSUE TITLE
Crash - sp_tabread_compute

### ISSUE THREAD

**Issue by craigory623** [AudioKit/AudioKit#1915]
Hello, 

  I consistently get crash reports in Xcode (currently 11.2 with AudioKit 4.9.2) of sp_tabread_compute from within tabled.c file

`#0	0x0000000104ed9a68 in sp_tabread_compute at /Xcode/AudioKit-master/AudioKit/Core/Soundpipe/modules/tabread.c:39
`

`#1	0x0000000104f2c968 in __43-[AKWaveTableAudioUnit internalRenderBlock]_block_invoke at /Xcode/AudioKit-master/AudioKit/Common/Nodes/Playback/Samplers/Wave Table/AKWaveTableAudioUnit.mm:157
`

I would appreciate any insight...

**Comment by aure** [comment]
This is not a lot to go on. Do you have some code we can see or ideally a sample project on Github that demonstrates this issue. AKWaveTable is put through its paces on a daily basis by Jeff, so I am sure he would have noticed this. Is this only with Xcode11.2 and AK4.9.2 or did you have it before this latest release of both?

**Comment by craigory623** [comment]
Sorry, because the crash was seemingly coming from different code than my own, I wasn't sure what to post. I have been getting a fair amount of these crashing for the better part of a year. 
My app basically uses the stock pitch tracking code for AKAnalyzer. This is my singleton class for audio kit. 

`class AudioCore {
    static let sharedInstance = AudioCore()
    
    let bgFile:AKAudioFile!
    let microphone:AKMicrophone!
    var analyzer:AKFrequencyTracker!
    let backgroundMusic:AKWaveTable!
    let buttonSound:AKWaveTable!
    let startSound:AKWaveTable!
    let stopSound:AKWaveTable!
    let explodeSound:AKWaveTable!
    
    init() {
        
        AKSettings.audioInputEnabled = true
        let sSound = try? AKAudioFile(readFileName: "Bleep_5.wav")
        startSound = AKWaveTable(file: sSound!)
        let stSound = try? AKAudioFile(readFileName: "stop.wav")
        stopSound = AKWaveTable(file: stSound!)
        let eSound = try? AKAudioFile(readFileName: "Bleep_4.wav")
        explodeSound = AKWaveTable(file: eSound!)
        bgFile  = try? AKAudioFile(readFileName: "intro.mp3")
        backgroundMusic  = AKWaveTable(file: bgFile!)
        backgroundMusic.volume = 0.3
        let selectSound  = try? AKAudioFile(readFileName: "Bleep_1.wav")
        buttonSound  = AKWaveTable(file: selectSound!)
        buttonSound.volume = 1
        microphone = AKMicrophone()
        microphone.volume = 0.5
        analyzer =  AKFrequencyTracker(microphone)
        let silence = AKBooster(analyzer,gain: 0.0)
        let soundMix = AKMixer(backgroundMusic,buttonSound,startSound,stopSound,explodeSound)
        let allMixer = AKMixer(soundMix, silence)
        try! AKSettings.session.overrideOutputAudioPort(AVAudioSession.PortOverride.speaker)
        AudioKit.output = allMixer
        startAudioEngine()
      
    }
    internal func startAudioEngine() {
        do {
            try AudioKit.start()
        } catch {
            AKLog("ERROR AudioKit did not start")
        }
    }
    
    internal func stopAudioEngine() {
        do {
            try AudioKit.stop()
        } catch {
            AKLog("ERROR AudioKit did not stop")
        }
    }
    internal func goToBackgroud() {
        startSound.stop()
        stopSound.stop()
        explodeSound.stop()
        backgroundMusic.stop()
        buttonSound.stop()
        microphone.stop()
        analyzer.stop()
        
    }
    internal func comeToFront() {
        
        microphone.start()
        analyzer.start()
    }
    
    
}`

This code to detect pitch is below
`@objc func updateUI() {
        if audioCore.analyzer!.amplitude > 0.05 {
            let fudge:Float = Float(audioCore.analyzer!.frequency * pitchChange)
            let newFrequency = audioCore.analyzer!.frequency + fudge
            frequencyLabel.text = String(format: "%0.1f", newFrequency)
            var frequency:Float = Float(newFrequency)
            playedFrequency = frequency
            while frequency > Float(noteFrequencies[noteFrequencies.count - 1]) {
                frequency /= 2.0
            }
            while frequency < Float(noteFrequencies[0]) {
                frequency *= 2.0
            }
            var minDistance: Float = 10000.0
            var index = 0
            for i in 0..<noteFrequencies.count {
                let distance = fabsf(Float(noteFrequencies[i]) - frequency)
                if distance < minDistance {
                    index = i
                    minDistance = distance
                }
            }
            let octave = Int(log2(newFrequency / frequency))
            noteName = String(format: "%@%d", noteNamesWithSharps[index], octave, noteNamesWithFlats[index], octave)
             displayLabel.text = String(format: "%@%d", displayedNotes[index],octave)
            noteNameLabel.text = noteName
        } else{
            displayLabel.text = ""
            frequencyLabel.text = "---"
            noteNameLabel.text = "---"
            playedFrequency = 0.0
            noteName = ""
        }
        amplitudeLabel.text = String(format: "%0.2f", audioCore.analyzer!.amplitude)
       
    }`

**Comment by trevor-sonic** [comment]
I have the same crash issue apparently.
I posted here the screenshot of the Crashlytics console.
For the moment I can't provide code.
I hope this helps.

![Screenshot 2019-11-21 at 20 17 12](https://user-images.githubusercontent.com/307560/69373537-edf5e200-0c9b-11ea-9c7d-881362adad57.png)

**Comment by aure** [comment]
I have recently encountered this error.Here is how I solved it. For me, it happened when I was coming in and out handling incoming phone calls. I was sequencing and I stopped the sequencer and then tried to stop AudioKit before handing the audio back to the system for the phone call. However, I neglected to stop the wave table sampler, so I ran into the table reading issue. Once I stopped the wave table as well, the problem went away.  I can see from the first code provided in this issue that craigory623 was more diligent about the stopping than I was. Two other points, make sure if you are handling route changing yourself, to set the AKSettings.enableRouteChangeHandling = false. Also, instead of doing a full AudioKit.stop, you might be better served by and AudioKit.engine.pause and AudioKit.engine.start.
